segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/BaseMetadataStore.java

```diff
@@ -636,11 +645,15 @@ private void releaseKeys(MetadataTransaction txn) {
     private CompletableFuture<TransactionData> loadFromStore(String key) {
         log.trace("Loading key from the store key = {}", key);
         return readFromStore(key)
-                .thenApplyAsync(this::makeCopyForBuffer, executor)
-                .thenApplyAsync(copyForBuffer -> {
-                    Preconditions.checkState(null != copyForBuffer, "Copy for buffer must not be null.");
-                    Preconditions.checkState(null != copyForBuffer.getDbObject(), "Missing tracking object");
-                    return insertInBuffer(key, copyForBuffer);
+                .thenApplyAsync(copyFromStore -> {
+                    Preconditions.checkState(null != copyFromStore, "Data from table store must not be null.");
```

Comment:
Is there more info we can put in these messages? Is `key` something useful?

Reply:
fixed.